tests/functional/flakes/develop.sh

```diff
@@ -43,13 +43,92 @@ echo "\$ENVVAR"
 EOF
 )" = "a" ]]
 
-# Test whether `nix develop --ignore-environment` does _not_ pass through environment variables.
+# Test whether `nix develop --ignore-env` does _not_ pass through environment variables.
 [[ -z "$(
-    ENVVAR=a nix develop --ignore-environment --no-write-lock-file .#hello <<EOF
+    ENVVAR=a nix develop --ignore-env --no-write-lock-file .#hello <<EOF
 echo "\$ENVVAR"
 EOF
 )" ]]
 
+# Test wether `--keep-env-var` keeps the environment variable.
+(
+  expect='BAR'
+  got="$(FOO='BAR' nix develop --ignore-env --keep-env-var FOO --no-write-lock-file .#hello <<EOF
+echo "\$FOO"
+EOF
+)"
+  [[ "$got" == "$expect" ]]
+)
+
+# Test wether `--set-env-var` sets the environment variable.
+(
+  expect='BAR'
+  got="$(nix develop --ignore-env --set-env-var FOO 'BAR' --no-write-lock-file .#hello <<EOF
+echo "\$FOO"
+EOF
+)"
+  [[ "$got" == "$expect" ]]
+)
+
+# Test wether multiple `--unset-env-var` and `--set-env-var` cancle out
+(
+  expect=''
+  got="$(nix develop --set-env-var FOO 'BAR' --unset-env-var FOO --no-write-lock-file .#hello <<EOF
+echo "\$FOO"
+EOF
+)"
+  [[ "$got" == "$expect" ]]
+)
+
+# Test that `--unset-env-var` unsets the variable, regardless of where in the command it is.
```

Comment:
I think this should be an error, as it is quite likely that the user made a mistake by both setting and unsetting a variable.
Changing it to an error may also let us change the semantics to first-wins or last-wins in the future.

Reply:
We also have a similar ambiguity in the `--ignore-environment` `--keep-var` situation as well.

Reply:
For the `--ignore-environment` `--keep-var` case, we have this:

```cpp
    if (!ignoreEnvironment && !keepVars.empty())
        throw UsageError("--keep-env-var does not make sense without --ignore-env");
```